### PR TITLE
Update the modal design

### PR DESCRIPTION
**Pull request by jameskoster**
Our modals look a little tired:

https://user-images.githubusercontent.com/846565/166428245-d920d3fa-a514-4f0c-a3d8-a803ed3fd5f2.mp4

This PR tries a few adjustments:

* Encourage focus on the modal contents by blurring the background
* Header styling:
  * Reduce noise (underline)
  * Adjust hierarchy – larger heading and close icon 
* Softens the corners

https://user-images.githubusercontent.com/846565/166428706-61094e69-9b1e-4bed-9152-b951a0d69b73.mp4

Perhaps some of these changes are more successful than others. Or perhaps none of them are :) What do you think?

### CODE REVIEW

File: packages/components/src/modal/style.scss
Comment (jameskoster) [inline packages/components/src/modal/style.scss:81]: It would be nice for things like this to be systematised, I'm aware of the experimental `Text` component, but it's not widely used. Perhaps there's a better way?

**Comment by stokesman** [comment]
I tested this on Gutenberg.run. Here's the welcome modal:
image

While I'm sure the right decisions will be reached here regardless of my input, here are my opinions. I think the heading size change succeeds greatly. I'm not sure about removing the border. Though I like it at a glance, I don't love the lack of a defined edge to scrolling content:
image

I think the blur is a win. The extra roundness on the corners seems alien.

Great to see the work on this 👍

**Review by apeatling — COMMENTED** [review]
A few thoughts:

* I think the blurring of the background is a big improvement in terms of modal focus, this is assuming there's no impact on performance (window resize, scrolling).
* I think the larger title is clearer.
* The missing under title border makes scrolling less clear. If this could appear once scrolled I think that could be a good compromise.
* The rounded corners -- are there any other examples of that size radius in Gutenberg? I like the look, but it seems like it departs from the current overall aesthetic.

**Comment by pablohoneyhoney** [comment]
> The rounded corners -- are there any other examples of that size radius in Gutenberg? I like the look, but it seems like it departs from the current overall aesthetic.

I agree with this and ideally we could align the radii to the intent of the rest of the current, already refreshed system.

**Comment by jameskoster** [comment]
Thanks for all the reviews :)

> intent of the rest of the current, already refreshed system

Is this documented anywhere, or is it just a case of always using a 2px radius?

My rationale for updating it is to harmonise nested radii. I feel an awkward tension when there's no scale.

| Tension | Harmony |
| --- | --- |
| Screenshot 2022-05-04 at 11 33 46 | Screenshot 2022-05-04 at 11 32 56 |

> this is assuming there's no impact on performance

Yes this needs testing on a variety of setups. It feels fine on my 2019 MacBook Pro, but obviously that's a relatively capable machine.

> If this could appear once scrolled I think that could be a good compromise.

Yeah that would be nice, would there be an impact on performance?

**Comment by pablohoneyhoney** [comment]
> My rationale for updating it is to harmonise nested radii. I feel an awkward tension when there's no scale.

imo I'd argue the opposite is true based on the images above. It doesn't need to be 2 in all applications, but certainly less rounded, retaining a bit more uniqueness and less trendiness.

**Comment by jameskoster** [comment]
Just to clarify, it is a suggestion motivated by maths rather than trends :D 

radii

I'll revert for now though, this is probably something to look at more holistically.

**Comment by pablohoneyhoney** [comment]
:D 

I think it's more to do with general design direction. 
There's certainly room to look at this more holistically, agreed, even trying 2/4 in cases like this.

**Comment by ciampo** [comment]
Thank you @jameskoster for working on this! The modal definitely needs a bit of visual polish.

Since this component is probably used in a few places across Gutenberg, we should make sure that the modal still looks good in all of these occurrences.

I would also suggest using Storybook when working on a component, as it helps us to first make sure that a given component works well in isolation, before seeing how it integrated in the block editor.

I also agree with the comments above re.:
- making sure we use the same border-radius as in other parts of the components library
- doing more testing on lower-end devices re. the blur effect — the `filter` CSS prop can require a lot of computation to render, especially when applied to the full viewport.

Finally, could you add an entry to the package's [CHANGELOG](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) ?

File: packages/components/src/modal/style.scss
Comment (ciampo) [reply packages/components/src/modal/style.scss:81]: I'd personally love to see the [`Text`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-text--default) and [`Heading`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-heading--default) components user more often.

I'm not sure if we need to first assess the type scale (and the rest of the styles), but otherwise I think we should slowly try to adopt them in other components, find out if they match our needs and make any necessary changes, before then rolling them out more widely across Gutenberg

**Comment by jameskoster** [comment]
Here's the Modal updates in Storybook:


https://user-images.githubusercontent.com/846565/167397478-054fc225-58b4-40f7-b628-b5d8c999b9ea.mp4

The background isn't blurred because I am targeting `.modal-open #wpwrap`, perhaps it would be better to simply target `body.modal-open`?

**Comment by mtias** [comment]
I don't think it makes sense to try to blur all the contents. Can it use `backdrop-filter` instead?

**Comment by jameskoster** [comment]
It's an option, but [browser support](https://caniuse.com/?search=backdrop-filter) isn't as good.

**Comment by jameskoster** [comment]
I suppose it doesn't hurt to consider the blur a progressive enhancement, the performance improvements are probably worth it.

**Comment by ciampo** [comment]
Hey @jameskoster , I took the liberty of implementing one of the requested features where the border only shows once there is scrolled content

https://user-images.githubusercontent.com/1083581/170286934-3103de2c-1ca5-41bd-a2dd-88f71d2b8cdc.mp4

 Feel free to take it from here and tweak the design further :)

File: packages/components/src/modal/stories/index.js
Comment (ciampo) [inline packages/components/src/modal/stories/index.js]:
```suggestion
						Lorem ipsum dolor sit amet, consectetur adipiscing
```

File: packages/components/src/modal/stories/index.js
Comment (ciampo) [inline packages/components/src/modal/stories/index.js]:
```suggestion
						anim id est laborum.
```

**Comment by jameskoster** [comment]
Thanks @ciampo :)

I just pushed a small update to adjust the alignment of the header content, and to better position the border that appears on scroll:


https://user-images.githubusercontent.com/846565/170314354-26a19970-17c6-4287-bd94-f98effd88973.mp4

**Comment by ciampo** [comment]
I pushed a couple more commits, simplifying the scroll event listener implementation and fixing the lint issues that were preventing other CI tasks from running.

I also had a brief look at the `backdrop-filter` animation on my high-end machine (with CPU slowdown x6), the animation 
 still run smoothly (although I know this is just an approximation of a real-world, lower-end device).

Thinking about what's left to do before this PR gets merged:

- make sure that all feedback has been addressed (I'm a bit lost on what we should do about border radii)
- thoroughly check all modals across Gutenberg, making sure that they still look good
- anything else?

**Comment by jameskoster** [comment]
> I'm a bit lost on what we should do about border radii

I already reverted the radii :) 

> thoroughly check all modals across Gutenberg, making sure that they still look good

Next week I'll screenshot all of our modals and report back.

**Comment by jameskoster** [comment]
Incoming screenshots. This is everything I found after a search for `<Modal` in the Gutenberg codebase.

### Editor
**Keyboard shortcuts**
pizzaplanet local_wp-admin_post php_post=1 action=edit

**New template**
pizzaplanet local_wp-admin_post php_post=1 action=edit (1)

**Pattern explorer**
pizzaplanet local_wp-admin_post-new php_post_type=page

**Choose a template part**
pizzaplanet local_wp-admin_themes php_page=gutenberg-edit-site postType=page postId=2

**Rename template**
pizzaplanet local_wp-admin_themes php_page=gutenberg-edit-site postType=page postId=2 (1)

**This post is being edited**
Screenshot 2022-05-31 at 11 57 44

**Someone has taken over this post**
pizzaplanet local_wp-admin_post php_post=1 action=edit (2)

**Preferences**
pizzaplanet local_wp-admin_post php_post=1 action=edit (3)

**Create reusable block**
pizzaplanet local_wp-admin_post php_post=1 action=edit (4)

**Guide**
pizzaplanet local_wp-admin_post php_post=1 action=edit (5)

### Navigation (beta)
**New menu**
pizzaplanet local_wp-admin_admin php_page=gutenberg-navigation

**Manage locations**
pizzaplanet local_wp-admin_admin php_page=gutenberg-navigation (1)

---

In some cases the spacing between the header text and the modal content feels excessive. I'll adjust that when I get a spare moment :)

**Comment by jameskoster** [comment]
I removed the margin applied to the pseudo element in `.components-modal__content`. As far as I can see this was only needed to create space beneath the bottom border on the modal header, but since we don't have that anymore (until scroll) it doesn't seem necessary. This facilitated some further tidy up as well since several modal instances were already overriding this behaviour. Screenshots:

pizzaplanet local_wp-admin_admin php_page=gutenberg-navigation
pizzaplanet local_wp-admin_post php_post=1 action=edit (1)
pizzaplanet local_wp-admin_post php_post=1 action=edit (6)
pizzaplanet local_wp-admin_post php_post=1 action=edit
pizzaplanet local_wp-admin_themes php_page=gutenberg-edit-site postType=page postId=2

**Comment by jameskoster** [comment]
One last thing to fix – the controls on the query replacement modal. Seems to be due to the explicit height. 

image

**Comment by ciampo** [comment]
We should also check components depending on `Modal`, like [`ConfirmDialog`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-confirmdialog--default) (@chad1008 may be able to help to check its usage across Gutenberg) and [`Guide`](https://wordpress.github.io/gutenberg/?path=/story/components-guide--default) (although I've never worked on `Guide` before).

**Comment by jameskoster** [comment]
I fixed the cut-off actions in the query pattern selector:

pizzaplanet local_wp-admin_themes php_page=gutenberg-edit-site postType=wp_template postId=twentytwentytwo%2F%2Findex

Though I notice the border-on-scroll behaviour isn't being applied because the border is added to `.components-modal__content` which isn't present here. Maybe it would be better to add the border to `.components-modal__header` @ciampo?

> and [Guide](https://wordpress.github.io/gutenberg/?path=/story/components-guide--default)

Guide seems to be fine: 

Screenshot 2022-05-31 at 16 49 05

[`ConfirmDialog`](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-confirmdialog--default) seems to be fine as well:

pizzaplanet local_wp-admin_post php_post=1 action=edit

**Review by ciampo — COMMENTED** [review]
> Though I notice the border-on-scroll behaviour isn't being applied because the border is added to `.components-modal__content` which isn't present here. Maybe it would be better to add the border to `.components-modal__header` @ciampo?

We can't add it to the header, because it's not guaranteed to be always displayed (see the `__experimentalHideHeader` prop).

Can you write instructions on how to display this particular instance of the modal that you're referring to? From a quick look, the element with class `.components-modal__content` should be always present in the DOM.

_(also a heads-up that, for the next week, I may slow to respond as I travel to and attend WCEU)_

**Comment by jameskoster** [comment]
> We can't add it to the header, because it's not guaranteed to be always displayed (see the __experimentalHideHeader prop).

Perhaps I am being shortsighted, but if the header isn't displayed then we don't need the border anyway because there will be nothing between the content and the boundary of the modal? 🤔 

> Can you write instructions on how to display this particular instance of the modal that you're referring to? 

Apologies. Here you go:

* Insert Query Loop in a post or template
* Click "Choose" in the block placeholder

> From a quick look, the element with class .components-modal__content should be always present in the DOM.

On closer inspection you're correct, but the scroll class isn't being applied:

https://user-images.githubusercontent.com/846565/171227861-26232f7d-74d4-4d37-bc75-f6b20810e4db.mp4

This seems to be due to the fact that `.block-editor-block-pattern-setup__grid` is the element that scrolls rather than  `.components-modal__content`.

Edit: It looks relatively easy to fix, but I'm unsure whether it should be a part of this PR or handled separately?

**Comment by ciampo** [comment]
> This seems to be due to the fact that `.block-editor-block-pattern-setup__grid` is the element that scrolls rather than `.components-modal__content`.

I would argue that this specific instance should be tweaked in order to make the `.components-modal__content` the scrolling element

> Edit: It looks relatively easy to fix, but I'm unsure whether it should be a part of this PR or handled separately?

I would handle it in a separate PR — let's keep this PR as focused as possible on the `Modal` component itself.

> Perhaps I am being shortsighted, but if the header isn't displayed then we don't need the border anyway because there will be nothing between the content and the boundary of the modal? 🤔

That's a fair point — feel free to apply the border styles to the header element (although it won't solve the specific issue that you flagged about the `.block-editor-block-pattern-setup__grid` being the scrolling element in that specific instance).

**Comment by stokesman** [comment]
Just making sure this one wasn't left out. Looks fine to me 👍 
Screen Shot 2022-06-02 at 11 57 38
(Tested in Firefox, so no `backdrop-filter`).

**Comment by jameskoster** [comment]
I think this is ready. Pinging @WordPress/gutenberg-design for a look at details, like the blur value.

**Comment by paaljoachim** [comment]
I am testing the PR Gutenberg build.


### Preferences
Before:
Screen Shot 2022-06-06 at 13 08 15

Focus is somewhat guided to the modal which is on top of the content. Both have the same in focus sharpness.

After:
Screen Shot 2022-06-06 at 13 09 10

One thing I notice right away is that the current amount of blur removes any connection with the page one is on.
Lessening the blur will still keep a connection with the page and focus will be brought to where the page is in focus.

Using blur to guide the users attention.
Blurring out something even a weak blur will guide the user to what is in focus.
Finding the sweet spot in just enough blur to make it subtile and yet guiding the user will be helpful. As one does not lose connection with what the blurred area covers. One can still make out elements even when they are blurred will help in keeping the connection with the page.

Remember subtile effects are better, as the contrast is not as huge going from one screen to another.
It can become exhausting for the eyes with too much contrast.
As we need to work on subtile changes that give hints to the user and softly guiding them into decisions that need to be made along the way. Less contrast and more subtile methods makes for a better overall experience.

**Comment by jameskoster** [comment]
Thanks @paaljoachim. Perhaps a 2px blur is better?

pizzaplanet local_wp-admin_themes php_page=gutenberg-edit-site postType=page postId=2

**Comment by paaljoachim** [comment]
Hey @jameskoster 
Yes that feels better.

File: packages/block-editor/src/components/block-pattern-setup/index.js
Comment (ciampo) [inline packages/block-editor/src/components/block-pattern-setup/index.js:178]: Thank you for adding an inline comment to explain how the `height` is computed here!

I wonder if there's any way we could avoid using these hardcoded numbers altogether and use either variables, or a layout that intrinsically adopts the correct height.

File: packages/components/CHANGELOG.md
Comment (ciampo) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `Modal` design adjustments: Blur elements outside of the modal, increase modal title size, use larger close icon, remove header border when modal contents are scrolled. ([#40781](https://github.com/WordPress/gutenberg/pull/40781)).
```

**Review by ciampo — APPROVED** [review]
Code changes LGTM (apart from a couple of minor comments that I left).

The issue with the border not disappearing when scrolling the contents on the Query pattern selector modal can be dealt with in a separate PR (as [discussed previously](https://github.com/WordPress/gutenberg/pull/40781#issuecomment-1144995892)).

I'll let the rest of the folks also give a final look to the instances of this redesigned Modal components across Gutenberg and approve this PR as they see fit.

File: packages/block-editor/src/components/block-pattern-setup/index.js
Comment (jameskoster) [reply packages/block-editor/src/components/block-pattern-setup/index.js:178]: I agree. Perhaps we can look into that when we address the border issue in this particular modal.

**Comment by jameskoster** [comment]
On second thoughts, a more subtle blur kind of makes the effect feel unintentional. If you're able to easily read the content behind the modal, then the increased focus is less successful. I reverted the blur value. We can always adjust later based on broader feedback.

**Comment by apeatling** [comment]
I agree that the blur is better with the $grid-unit value, the change made the modal view much more noisy. I disagree more blur disconnects the user from the previous page and where they were. They just clicked on something that initiated this modal, and the pattern is extremely common across the web.

**Comment by javierarce** [comment]
I'm in favor of all the improvements and refreshments made to the modal, except for the blur. 

Maybe it's just its novelty and just me, but it doesn't feel attuned to the rest of the design. And I'm not sure it's really needed. Also, in one important instance, using the Preferences modal, things get a bit weird: you change the settings and some blurred and undefined elements appear or disappear on the back. Paradoxically, the blur makes things more distracting than in the previous version. 

https://user-images.githubusercontent.com/4933/173548101-0bb49277-ea64-4917-ad4d-c934c1b89f5c.mov

**Comment by critterverse** [comment]
I like the blur — it helps obscure the awkward overlapping effect that happens sometimes when you open a modal from a popover and the popover remains open in the background (like with the locking modal). My personal preference on the blur is that less is more! ~4px looks about right to me:

**Comment by noisysocks** [comment]
Just a quick note that we have a few components (pictured below) that contain CSS which removes the border underneath the heading. Since this PR removes this border for **all** modals, we can delete this CSS. See https://github.com/WordPress/gutenberg/pull/41998/commits/069dddb76e467ceaed8463c748849674403b2a82 for a hint of what changes to make.

| Welcome guide | Template rename | Create custom template | Page pattern picker |
| - | - | - | - |
| Screen Shot 2022-06-28 at 17 33 22 | Screen Shot 2022-06-28 at 17 33 24 | Screen Shot 2022-06-28 at 17 33 28 | Screen Shot 2022-06-28 at 17 33 30 |

**Comment by jameskoster** [comment]
Thanks @noisysocks, looks like a couple of those were missed, I'll push an update.

**Review by noisysocks — COMMENTED** [review]
Code LGTM! I fixed some merge conflicts.

I tested all of the modals mentioned in the comments of this PR in both Chrome (has blur) and Firefox (no blur). The only issue I spotted is this one that was already mentioned earlier:

> Though I notice the border-on-scroll behaviour isn't being applied because the border is added to .components-modal__content which isn't present here. Maybe it would be better to add the border to `.components-modal__header` @ciampo?

Would be good to fix that as it's a regression compared to what's in `trunk`.

Agree with @critterverse that the blur is especially good in situations where there is a modal on top of a popover. I also think the 4px blur looks nicer than 8px but not going to fight anyone about it.

**Comment by paaljoachim** [comment]
The tricky part here is will blur be added to all modals that open on top of the screen?
In general it feels that some modals might need a blurred background while other modals do not. Some modals it is nice to have these connected with the rest of the screen (as in no blur) and other modals do not need this connection with the background. But then which modals need a blur and which modals do not...?

Using blur is something that should not be taken lightly and should not often be used. As it can quickly get tiring for the eyes having the background blur when opening a modal. The various modals that @noisysocks Robert mentions above give a nice general UX overview of how modals can look when opened. A darker background is seen, but no blur effect. As the modal is highlighted because of the darker background. One can still have a relation of the modal to the background.

Bottom line for me as I view it: Instead of a blurred background, what about using the effects that @noisysocks shows in the above screenshots? https://github.com/WordPress/gutenberg/pull/40781#issuecomment-1173256904
As it gives a nice highlight of the modal while adding a subtile darkening of the background. It is a softer approach compared to using a blurred effect.

Reading this:

> Agree with @critterverse that the blur is especially good in situations where there is a modal on top of a popover. I also think the 4px blur looks nicer than 8px but not going to fight anyone about it.

We could perhaps try it out. I am not certain about it, and I am wondering if we should stick to one UX visual method when opening a modal.

Regarding Lock Cover modal which @critterverse shows above. Here is a situation where it is better to also see the connected background (no blur). Locking something I want to see the elements that I plan on locking. Keeping the connection in place. When a modal is connected with one or more elements in the background it is important to be able to visually shift between foreground (modal) and background (element being affected) to decide options one wants to click in the modal. Blurring the background one would not be able to shift focus to the background to help decide which options to click in the modal.

**Comment by jameskoster** [comment]
> When a modal is connected with one or more elements in the background it is important to be able to visually shift between foreground (modal) and background (element being affected) to decide options one wants to click in the modal

I don't know if this is a heuristic we should establish as a guiding principle, it sets a precedent that is going to be difficult to follow consistently. For example on a small screens modals occupy the entire viewport, totally obscuring the rest of the UI. Encouraging folks to concentrate on both the modal contents and the elements behind it also feels like kind of a bad practise... doesn't this defeat the purpose of using a modal in the first place? 

If there's an option that relies on the visibility of the broader interface then it probably shouldn't be in a modal, but that's another issue altogether. To re-iterate modal principles from the [handbook](https://developer.wordpress.org/block-editor/reference-guides/components/modal/):

> * Focused. Modals pull user attention away from the rest of the screen to focus their attention, ensuring that the modal’s content is addressed.
> * Direct. Modal text should communicate important information and be dedicated to helping the user appropriately complete a task.
> * Helpful. Modals should appear in response to a user task or an action to offer relevant and contextual information

---

We're going in circles a little bit here, so if I may propose a next step: why don't we merge this PR, test it with a wider audience as a part of the plugin, and revert / refine if necessary? There are other improvements to the modal included here that would be nice to deploy. What do y'all think?

File: packages/components/src/modal/style.scss
Comment (ciampo) [inline packages/components/src/modal/style.scss:121]: Just flagging that this change may have introduced other layout regressions in places where it was assumed that the Modal's header was `$header-height` px tall.

One such example is the block pattern inserter/explorer modal, where in my testing I spotted a regression caused by the fact that the sidebar was positioned at `$header-height` px from the top of its container

https://github.com/WordPress/gutenberg/blob/8b1492131355aa6d42272faba25b841f0cd4509d/packages/block-editor/src/components/inserter/style.scss#L345

image

In general, consumers of the `Modal` component should not assume implementation details like the height of its header — when that happens, a small change like this PR is enough to cause a regression (which is also what I was trying to flag in [this other comment of mine](https://github.com/WordPress/gutenberg/pull/40781/files#r893476058) ).

While a short-term fix would be to update the line of code linked above to use the updated Modal's header height, I would argue that the best way to fix this specific regression in the block pattern inserter / explorer is to rewrite the modal's internal layout in a way that avoids absolute positioning and hardcoded margins/positions.

**Review by ciampo — COMMENTED** [review]
> > Though I notice the border-on-scroll behaviour isn't being applied because the border is added to `.components-modal__content` which isn't present here. Maybe it would be better to add the border to .components-modal__header @ciampo?
> 
> Would be good to fix that as it's a regression compared to what's in trunk.

That regression is actually caused by the fact that, specifically for the Query pattern selector block, some custom styles are preventing the modal's content wrapper from scrolling (and therefore never triggers the logic to show/hide the border):

https://github.com/WordPress/gutenberg/blob/6d1496a88b92c751ef5322361a53b889ab97c03e/packages/block-library/src/query/editor.scss#L22-L28

To fix the regression, that modal should be rewritten in a such a way that allows the `.components-modal__content` element to scroll.

Regarding whether the border should be applied to the "header" or the "content" wrapper, that's a matter of preference and, in my opinion, more of an implementation detail.

File: packages/components/src/modal/style.scss
Comment (jameskoster) [reply packages/components/src/modal/style.scss:121]: > I would argue that the best way to fix this specific regression in the block pattern inserter / explorer is to rewrite the modal's internal layout avoiding absolute positioning and hardcoded distances.

I agree, the hardcoded height also causes issues where the modal title is long enough to wrap (uncommon but not impossible). Do you think we should do this as a part of this PR or elsewhere?

Comment (ciampo) [reply packages/components/src/modal/style.scss:121]: Probably the best course of action is to:

- add a quick fix in this PR (should be as easy as changing from `top: $header-height;` to `top: $header-height + $grid-unit-20;`
- open a few follow-up PRs to:
  - refactor the layout of the component as suggested before.
  - try to refactor away from the hardcoded height as discussed in [this other comment](https://github.com/WordPress/gutenberg/pull/40781/files#r893476058)

**Comment by apeatling** [comment]
> We're going in circles a little bit here, so if I may propose a next step: why don't we merge this PR, test it with a wider audience as a part of the plugin, and revert / refine if necessary? There are other improvements to the modal included here that would be nice to deploy. What do y'all think?

+1 on merging and refining after the above quick fix. I don't think we need to bike shed on the amount of blur and hold this up. :+1:

**Review by apeatling — APPROVED** [review]

**Comment by annezazu** [comment]
>why don't we merge this PR, test it with a wider audience as a part of the plugin, and revert / refine if necessary? 

This is on my list to have as part of a broader call for testing for the FSE Outreach program to hopefully get some feedback for what it's worth (been following this PR).

**Comment by jameskoster** [comment]
Pushed an update to fix the issue above:

Screenshot 2022-07-06 at 10 19 17

Will merge once tests pass :)

**Comment by ciampo** [comment]
Just to summarize the follow-up work that we agreed during this review:

- **Query pattern selector block modal**: refactor the layout to allow the modal's content wrapper to scroll ([more details](https://github.com/WordPress/gutenberg/pull/40781#pullrequestreview-1028437034)) (this is the most urgent task, since the current PR may have [introduced a regression](https://github.com/WordPress/gutenberg/pull/40781#pullrequestreview-1027968436))
- **Block pattern inserter/explorer's modal**: refactor the layout of the component so that it doesn't rely on absolute positioning and hardcoded modal header height ([more details](https://github.com/WordPress/gutenberg/pull/40781#discussion_r913840168))
- **Block pattern setup modal**: refactor the layout of the component so that it doesn't rely on hardcoded modal header heights ([more details](https://github.com/WordPress/gutenberg/pull/40781#discussion_r893476058))

**Comment by jameskoster** [comment]
I opened https://github.com/WordPress/gutenberg/pull/42194 to address the first issue, and https://github.com/WordPress/gutenberg/issues/42195 to address the latter two.

**Comment by andrewserong** [comment]
I really like the smooth look of the blurred background on the modal, but I just encountered a potential usability issue where there is an element outside of the modal that is contextually related to the content of the modal. For example, if you go to adjust an item in the List View to apply or remove a Lock, with the modal open, you can no longer see which (particular) block you have highlighted:

image

I'm curious if there are other cases where we have this sort of contextual issue — perhaps in the case of Lock, the use of a Modal isn't quite right, maybe it should be a Popover instead in order to better highlight the relationship to the currently selected item. (I'm happy to open up a separate issue if folks think the approach for the Lock modal should be adjusted rather than the styling of the Modal).

**Comment by jameskoster** [comment]
As you alluded to, I think this could actually be a helpful guiding principle around when to (or not) use modals. I outlined in more detail [here](https://github.com/WordPress/gutenberg/pull/40781#issuecomment-1174819631), but generally speaking if you need to observe both the modal contents and the rest of the UI together, then a modal is probably not going to be the best choice.

That said, I don't find this particular instance to be problematic since the modal title clearly indicates which block you're locking.

**Comment by andrewserong** [comment]
Great, thanks for the explanation @jameskoster! Also, I suppose since modals are typically fullscreen on mobile, that's probably another reason to avoid a modal if it has to maintain a connection to information outside of the modal.

> That said, I don't find this particular instance to be problematic since the modal title clearly indicates which block you're locking.

Okie-doke, I'm quite possibly over thinking this one because I've been looking at the List View closely, so noticed a difference once I rebased against trunk and saw that the modal looked different. I'm happy to leave this, then, given that the block title is in the modal title 👍

**Comment by paaljoachim** [comment]
I have gone through the call for testing: https://make.wordpress.org/test/2022/07/11/fse-program-testing-call-15-category-customization/#comment-2608

I noticed the various modals that showed up and how blurred the background area around it is. 
Here it would be good to take a closer look at OSX, Windows etc and see how modals and the area around it is used.

In general for a user interface the key is having subtile changes that give hints along the way to user. The current blur is too much of a change and makes it harder on the eyes working in Gutenberg. In general one does not want to bring too much contrast when working in any software. Think if OSX or Windows where to blur the background every time a modal is seen that one has to relate to before moving on. If one tries to do something else one is not able to and will need to find the modal that one does need to relate to before moving on. 

I do look forward to hearing more view points from other folks about the blur effect seen in the screen of it seems any modal that is opened.

**Pull request merged**
